### PR TITLE
cleanup misleading errors in openAPI tesing

### PR DESCRIPTION
**Pull request by RainbowMango**
**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:
Cleanup misleading error handling.

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:
```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

/priority backlog

### CODE REVIEW

File: staging/src/k8s.io/client-go/discovery/discovery_client_test.go [hunk 1]
```diff
@@ -455,58 +455,77 @@ func returnedOpenAPI() *openapi_v2.Document {
...
 func openapiSchemaDeprecatedFakeServer(status int) (*httptest.Server, error) {
-	var sErr error
 	server := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, req *http.Request) {
 		if req.URL.Path == "/openapi/v2" {
 			// write the error status for the new endpoint request
 			w.WriteHeader(status)
 			return
 		}
 		if req.URL.Path != "/swagger-2.0.0.pb-v1" {
-			sErr = fmt.Errorf("Unexpected url %v", req.URL)
```
Comment (RainbowMango) [inline staging/src/k8s.io/client-go/discovery/discovery_client_test.go:466]: The `sErr` will never return to the client.

**Comment by RainbowMango** [comment]
/assign @liggitt

The failing test should be flak.

File: staging/src/k8s.io/client-go/discovery/discovery_client_test.go [hunk 2]
```diff
@@ -455,58 +455,77 @@ func returnedOpenAPI() *openapi_v2.Document {
 }
 
 func openapiSchemaDeprecatedFakeServer(status int) (*httptest.Server, error) {
```
Comment (liggitt) [inline staging/src/k8s.io/client-go/discovery/discovery_client_test.go:457]: would it make more sense to pass in `t *testing.T` and call `t.Errorf` for these cases to be sure they fail the test

Comment (RainbowMango) [reply staging/src/k8s.io/client-go/discovery/discovery_client_test.go:457]: To be honest, given the handler has more than one exit, the expectation of cases may be different, we'd better not coupling that.

On the other hand, every case's checkpoint mainly relies on the handler's output, thus if there is something wrong in the handler, they would find it.

**Comment by RainbowMango** [comment]
Hi @liggitt , I pushed a new commit to address your comments, could you please take a look again?

/label tide/merge-method-squash

**Pull request merged**